lib/Dialect/FIRRTL/FIRRTLOps.cpp

```diff
@@ -472,12 +467,16 @@ static LogicalResult verifyCircuitOp(CircuitOp circuit) {
                << expectedModuleName
                << " instead found: " << innerRef.getModule();
 
-      if (auto mod = dyn_cast_or_null<FModuleLike>(rec->op)) {
-        auto annos = AnnotationSet::forPort(mod, rec->portIdx);
-        leafFound = hasNonLocal(annos, nlaName);
+      if (auto mod = dyn_cast<FModuleLike>(rec->op)) {
+        leafFound =
+            hasNonLocal(AnnotationSet::forPort(mod, rec->portIdx), nlaName);
       } else if (rec->op) {
-        AnnotationSet annos(rec->op);
-        leafFound = hasNonLocal(annos, nlaName);
+        leafFound = hasNonLocal(AnnotationSet(rec->op), nlaName);
+        if (auto mem = dyn_cast<MemOp>(rec->op)) {
+          for (unsigned i = 0, e = mem.getNumResults(); !leafFound && i < e;
+               ++i)
+            leafFound = hasNonLocal(AnnotationSet::forPort(mem, i), nlaName);
```

Comment:
I think this will overwrite the `leafFound` for subsequent ports, so maybe break early if true.

Reply:
It's actually in the loop condition `!leafFound && i < e`

Reply:
of course, sorry missed it.